### PR TITLE
Hotfix: fix has_xxx_ownership's potential issues

### PR DESCRIPTION
**Pull request by henryyang42**
For `has_group_ownership`, `has_contest_ownership`, `has_problem_ownership`:
- call `validate_user` before using `User` object
- delete unnecessary judgements

### CODE REVIEW

File: utils/user_info.py [hunk 1]
```diff
@@ -21,73 +21,61 @@
...
+
 
 logger = get_logger()
 
 
-#contest ownership
 def has_contest_ownership(curr_user, curr_contest):
-    user_is_valid(curr_user) #check user
-    #check contset
-    try:
```
Comment (nickelcheng) [inline utils/user_info.py]: I think check whether contest exists or not is necessary.

File: utils/user_info.py [hunk 2]
```diff
@@ -21,73 +21,61 @@
...
-    return is_owner
+                return True
+    return False
 
 
-#group ownership
 def has_group_ownership(curr_user, curr_group):
-    user_is_valid(curr_user) #check user
-    #check group
-    try:
```
Comment (nickelcheng) [inline utils/user_info.py]: Ditto

File: utils/user_info.py [hunk 3]
```diff
@@ -21,73 +21,61 @@
...
-    return is_owner
+                return True
+    return False
 
 
-#problem ownership
 def has_problem_ownership(curr_user, curr_problem):
-    user_is_valid(curr_user) #check user
-    #check problem
-    try:
```
Comment (nickelcheng) [inline utils/user_info.py]: Ditto

**Comment by henryyang42** [comment]
The objects must have been checked some where else, it won't come out of nowhere though.

**Comment by nickelcheng** [comment]
I'm just afraid of bugs in other files (such as someone forgot to check contest existence and call has_contest_ownership directly).

**Comment by henryyang42** [comment]
Contest object must be fetched by ``Contest,objects.get()``. Should it fail, it will either ``raise exception``, or just crash. By no chance it will call that function if it does not exist.

**Comment by nickelcheng** [comment]
ok, then it's fine.

**Comment by nickelcheng** [comment]
+1, @bruce3557 please check.

File: utils/user_info.py [hunk 4]
```diff
@@ -144,18 +126,18 @@ def get_user_statistics(user):
     submissions = Submission.objects.filter(user=user)
     for label in status_labels:
         statistics += [{
-            'label': label,
-            'value': submissions.filter(status=label).count()
-        }]
+                           'label': label,
+                           'value': submissions.filter(status=label).count()
+                       }]
```
Comment (bruce3557) [inline utils/user_info.py]: prefer original indent, but you can discusse which one is better for your usage.

File: utils/user_info.py [hunk 5]
```diff
@@ -144,18 +126,18 @@ def get_user_statistics(user):
...
     submissions_id = map(lambda submission: submission.id, submissions)
     submission_details = SubmissionDetail.objects.filter(sid__in=submissions_id)
     for label in verdict_labels:
         statistics += [{
-            'label': label,
-            'value': submission_details.filter(verdict=label).count()
-        }]
+                           'label': label,
+                           'value': submission_details.filter(verdict=label).count()
+                       }]
```
Comment (bruce3557) [inline utils/user_info.py]: Ditto

Comment (henryyang42) [inline utils/user_info.py]: Corrected

**Comment by bruce3557** [comment]
+1

**Comment by nickelcheng** [comment]
going to merge.

**Pull request merged**